### PR TITLE
Support the VK_KHR_push_descriptor extension.

### PR DESCRIPTION
**Pull request by cdavis5e**

### CODE REVIEW

**Comment by billhollings** [comment]
@cdavis5e Nice work!

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 1]
```diff
@@ -77,6 +77,20 @@
     if (properties) {
         properties->sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2_KHR;
         properties->properties = _properties;
+        auto* next = (VkStructureType*)properties->pNext;
+        while (next) {
+            switch (*next) {
```
Comment (billhollings) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: @cdavis5e This line is causing an `EXC_BAD_ACCESS` error when running `vulkaninfo` from the 1.0.82 Vulkan SDK.

Comment (cdavis5e) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: I think I know what's wrong. I forgot to account for the padding between the `sType` and `pNext` members.

Comment (cdavis5e) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: #245

**Pull request merged**